debugger/src/main/java/org/codice/acdebugger/impl/Backdoor.java

```diff
@@ -0,0 +1,260 @@
+    "squid:S106", /* this is a console application */
+    "squid:S1148" /* this is a console application */
+  })
+  public synchronized void init(Debug debug, ObjectReference backdoorReference) {
+    if (backdoorReference == null) {
+      throw new IllegalStateException("unable to locate backdoor instance");
+    }
+    try {
+      this.initializing = true;
+      this.backdoorReference = backdoorReference;
+      this.getBundle =
+          debug.findMethod(
+              this.backdoorReference.referenceType(),
+              "getBundle",
+              "(Ljava/lang/Object;)Ljava/lang/String;");
+      this.getPermissionStrings =
+          debug.findMethod(
+              this.backdoorReference.referenceType(),
+              "getPermissionStrings",
+              "(Ljava/lang/Object;)Ljava/lang/String;");
+      this.grantPermission =
+          debug.findMethod(
+              this.backdoorReference.referenceType(),
+              "grantPermission",
+              "(Ljava/lang/String;Ljava/lang/String;)V");
+      this.hasPermission =
+          debug.findMethod(
+              this.backdoorReference.referenceType(),
+              "hasPermission",
+              "(Ljava/lang/Object;Ljava/lang/Object;)Z");
+      this.getServicePermissionInfoAndGrant =
+          debug.findMethod(
+              this.backdoorReference.referenceType(),
+              "getServicePermissionInfoAndGrant",
+              "(Ljava/lang/String;Ljava/lang/Object;Ljava/lang/Object;Z)Ljava/lang/String;");
+      System.out.println("AC Debugger: Backdoor discovered");
+    } finally {
+      this.initializing = false;
+    }
+    // now register the Eclipse service permission breakpoint, now that we know the policy has been
+    // loaded since the backdoor
+    if (debug.isMonitoringService()) {
+      try {
+        debug.add(new HasListenServicePermissionBreakpointProcessor());
+      } catch (Exception e) { // cannot register breakpoint so continue without it
+        e.printStackTrace();
+      }
+    }
+  }
+
+  /**
+   * Initializes the backdoor by attempting to find its instance in the attached VM.
+   *
+   * @param debug the current debug information
+   * @return <code>true</code> if the backdoor is initialized; <code>false</code> if not
+   */
+  public synchronized boolean init(Debug debug) {
+    if (initializing) {
+      return false;
+    } else if (backdoorReference == null) {
+      final ObjectReference ref =
+          debug
+              .classes(Backdoor.CLASS_SIGNATURE)
+              .map(clazz -> getInstance(debug, clazz))
+              .filter(Objects::nonNull)
+              .findFirst()
+              .orElse(null);
+
+      if (ref == null) {
+        return false;
+      }
+      init(debug, ref);
+    }
+    return true;
+  }
+
+  /**
+   * Gets a bundle location for the given object. The object can be a bundle, a classloader, a
+   * protection domain, a bundle context, or even a classloader. This methods makes all attempts
```

Comment:
Classloader is mentioned twice here. Was it meant to be something else or just a typo?

Reply:
Took care of those in the `Debug` class too 😄